src/3d/qgs3dutils.cpp

```diff
@@ -952,3 +956,25 @@ QByteArray Qgs3DUtils::removeDefinesFromShaderCode( const QByteArray &shaderCode
 
   return newShaderCode;
 }
+
+int Qgs3DUtils::getOpenGLMaxClipPlanes()
+{
+  int nrPlanes = 6;
+
+  QOffscreenSurface offscreen;
+  QOpenGLContext context;
+
+  offscreen.setFormat( QSurfaceFormat::defaultFormat() );
+  offscreen.create();
+  Q_ASSERT_X( offscreen.isValid(), Q_FUNC_INFO, "Unable to create offscreen surface to gather capabilities" );
```

Comment:
I believe asserts should only be used for cases where we're really sure they will not happen - creation of offscreen surface can fail if there's an issue with OpenGL drivers, and that should not crash the whole app...

Reply:
Removed.